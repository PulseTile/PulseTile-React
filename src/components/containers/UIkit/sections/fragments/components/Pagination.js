import React from 'react';
import PaginationBlock from '../../../../../presentational/PaginationBlock/PaginationBlock';

/**
 * This component returns content of Pagination section
 */
const Pagination = () => {
    return (
        <div id="pagination" className="ui-section">
            <strong className="ui-title">Pagination (page 18)</strong>
            <div className="row">
                <div className="col-xs-12 col-sm-6">
                    <div className="ui-sub-section">
                        <div className="ui-sub-title">Pagination normal</div>
                        <div className="wrap-control-group">
                            <div className="control-group with-indent left">
                                <PaginationBlock
                                    entriesPerPage={20}
                                    totalEntriesAmount={1200}
                                    offset={334}
                                    setOffset={function() {}}
                                />
                            </div>
                        </div>
                    </div>
                </div>
                <div className="col-xs-12 col-sm-6">
                    <div className="ui-sub-section">
                        <div className="ui-sub-title">Pagination short</div>
                        <div className="wrap-control-group">
                            <div className="control-group with-indent left">
                                <PaginationBlock
                                    entriesPerPage={20}
                                    totalEntriesAmount={19}
                                    offset={1}
                                    setOffset={function() {}}
                                />
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Pagination;
