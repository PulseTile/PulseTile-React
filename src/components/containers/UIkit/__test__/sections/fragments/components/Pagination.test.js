import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import Pagination from '../../../../sections/fragments/components/Pagination';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <Pagination />', () => {
  it('should renders Pagination with props correctly', () => {
    const component = shallow(<Pagination />);
    expect(component).toMatchSnapshot();
  });
});
