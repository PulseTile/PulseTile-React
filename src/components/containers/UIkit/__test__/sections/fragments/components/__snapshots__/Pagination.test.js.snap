// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Pagination /> should renders Pagination with props correctly 1`] = `
<div
  className="ui-section"
  id="pagination"
>
  <strong
    className="ui-title"
  >
    Pagination (page 18)
  </strong>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-sm-6"
    >
      <div
        className="ui-sub-section"
      >
        <div
          className="ui-sub-title"
        >
          Pagination normal
        </div>
        <div
          className="wrap-control-group"
        >
          <div
            className="control-group with-indent left"
          >
            <PaginationBlock
              entriesPerPage={20}
              offset={334}
              setOffset={[Function]}
              totalEntriesAmount={1200}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-sm-6"
    >
      <div
        className="ui-sub-section"
      >
        <div
          className="ui-sub-title"
        >
          Pagination short
        </div>
        <div
          className="wrap-control-group"
        >
          <div
            className="control-group with-indent left"
          >
            <PaginationBlock
              entriesPerPage={20}
              offset={1}
              setOffset={[Function]}
              totalEntriesAmount={19}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
